GitExtUtils/CredentialManager.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System.Net;
+
+namespace GitExtUtils
+{
+    public static class CredentialManager
+    {
+        private const string TargetPrefix = "GitExtensions_";
+
+        private static string GetTarget(string rawTarget)
+        {
+            return $"{TargetPrefix}{(rawTarget?.StartsWith(TargetPrefix) == true ? rawTarget.Replace(TargetPrefix, string.Empty) : rawTarget)}";
+        }
+
+        public static bool TryGetCredentials(string target, out NetworkCredential credentials)
+        {
+            credentials = AdysTech.CredentialManager.CredentialManager.GetCredentials(GetTarget(target));
+            return credentials != null;
+        }
+
+        public static bool SaveCredentials(string target, string userName, string password)
+        {
+            if (string.IsNullOrWhiteSpace(target))
+            {
+                return false;
+            }
+
+            return AdysTech.CredentialManager.CredentialManager.SaveCredentials(GetTarget(target), new NetworkCredential(userName.Trim(), password));
+        }
+
+        public static bool RemoveCredentials(string target)
+        {
+            if (string.IsNullOrWhiteSpace(target) || AdysTech.CredentialManager.CredentialManager.GetCredentials(GetTarget(target)) == null)
+            {
+                return false;
+            }
+
+            return AdysTech.CredentialManager.CredentialManager.RemoveCredentials(GetTarget(target));
+        }
+
+        public static bool UpdateCredentials(string target, string userName, string password)
```

Comment:
Should this be a [SecureString](https://docs.microsoft.com/en-us/dotnet/api/system.security.securestring?view=netframework-4.7.2)?
```suggestion
        public static bool UpdateCredentials(string target, string userName, SecureString password)
```

Reply:
Yes, it is more proper, but it does not make sense for Jira plugin, because there is not Jira API that take SecureString. If you wish I can add overloaded methods with SecureString. Also it does not make sense when users edit settings, because it requires special controls, which work with SecureString.

Reply:
That's cool, thank you for clarifying.